src/Gaffer/ArrayPlug.cpp

```diff
@@ -63,6 +64,8 @@ ArrayPlug::ArrayPlug( const std::string &name, Direction direction, PlugPtr elem
 		for( size_t i = 1; i < m_minSize; ++i )
 		{
 			PlugPtr p = element->createCounterpart( element->getName(), Plug::In );
+			MetadataAlgo::copyColorMetaData(element.get(), p.get());
```

Comment:
This is having the unfortunate side effect of adding a bunch of metadata where we don't need it. See the serialisation below :

```
import Gaffer
import GafferScene
import IECore

Gaffer.Metadata.registerNodeValue( parent, "serialiser:milestoneVersion", 0, persistent=False )
Gaffer.Metadata.registerNodeValue( parent, "serialiser:majorVersion", 32, persistent=False )
Gaffer.Metadata.registerNodeValue( parent, "serialiser:minorVersion", 0, persistent=False )
Gaffer.Metadata.registerNodeValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["Group"] = GafferScene.Group( "Group" )
parent.addChild( __children["Group"] )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in2", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = IECore.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"]["__uiPosition"].setValue( IECore.V2f( -10.4999266, -2.98233581 ) )
__children["Plane"] = GafferScene.Plane( "Plane" )
parent.addChild( __children["Plane"] )
__children["Plane"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = IECore.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Plane"]["__uiPosition"].setValue( IECore.V2f( -12.0000715, 6.18233585 ) )
__children["Group"]["in"]["in0"].setInput( __children["Plane"]["out"] )
__children["Group"]["in"]["in1"].setInput( __children["Plane"]["out"] )
Gaffer.Metadata.registerPlugValue( __children["Group"]["in"]["in1"], 'nodule:color', IECore.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )
Gaffer.Metadata.registerPlugValue( __children["Group"]["in"]["in2"], 'nodule:color', IECore.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )

del __children
```

I'm not sure how best to deal with this. Maybe we should query to see if the new plug already has colour metadata (in this case inherited from [here](https://github.com/GafferHQ/gaffer/blob/master/startup/gui/nodeGraph.py#L61)) and add it only if it doesn't? So perhaps `MetadataAlgo::copyColors( from, to, /* overwrite = */ false )`?